Signup/login pages

MilesDev91 opened the pull request:
### What this PR does (required):
- Styled the Signup and Login pages to match the mock! Changed the MUI theme which will affect the rest of the application.

### Screenshots / Videos (front-end only):
![Screenshot (40)](https://user-images.githubusercontent.com/34494229/148307405-65d5b515-9668-4af1-9957-dd5ae7b2aa02.png)

### Any information needed to test this feature (required):
- Everything should be responsive. Errors still function as normal. Click into the fields and try to submit with the wrong information.

### Any issues with the current functionality (optional):
- The forgot? text is still in blue. Will address (if needed) in the future.
- Wanted to have the forms prefilled (I think it looks nicer that way) but I was just short of figuring it out. Figured it was a better use of my time to continue and revisit in the future if I have time.

MilesDev91 committed:
Authheader and theme styling

Placed authheader below the form on both Signup and login.

Authheader now uses theme primary color instead of mui default.

Fixed styling and removed boxes so that the spacing and flow resembles the mock on both Signup and Login.

MilesDev91 committed:
Styling Changes

Reordered and changed the variants on the Mui text fields.

Attempting to fine-tune the styling on everything. Having trouble with MUI default overriding everything.

Centered some things on the page.

MilesDev91 committed:
Styling, styling, styling

Fixed import that was stopping theme injection.

Learned how to target root of text field class to override Material UI in scoped CSS where I don't want to override the entire theme. Changed styling and positioning to further match the mock.

Removed useTheme() from AuthHeader as that didn't address the "root" of the issue....also didn't work anyway

rajivtitus commented on client/src/pages/Login/Login.tsx:
```diff
@@ -38,25 +38,18 @@ export default function Login(): JSX.Element {
               <Grid item xs>
                 <Typography className={classes.welcome} component="h1" variant="h5">
-                  Welcome back!
+                  <strong>Welcome back!</strong>
```
Alternatively, you can apply a `font-weight` using the existing CSS class

MilesDev91 replied:
Ya this was before I figured out how to override MUI root classes without affecting the theme. I'll address this and other changes!

rajivtitus commented on client/src/pages/Login/LoginForm/LoginForm.tsx:
```diff
@@ -48,48 +48,47 @@ export default function Login({ handleSubmit }: Props): JSX.Element {
           <TextField
             id="email"
-            label={<Typography className={classes.label}>E-mail address</Typography>}
+            label={<Typography className={classes.label}>EMAIL ADDRESS</Typography>}
```
Please avoid using CAPITALCASE text as some screen readers interpret them as abbreviations. You can use a CSS class to transform the text case instead :)

Ditto throughout!

rajivtitus commented on client/src/pages/SignUp/SignUp.tsx:
```diff
@@ -39,25 +39,18 @@ export default function Register(): JSX.Element {
               <Grid item xs>
                 <Typography className={classes.welcome} component="h1" variant="h5">
-                  Create an account
+                  <strong>Sign up</strong>
```
Same as comment above about usage of a CSS class instead

rajivtitus commented on client/src/themes/theme.ts:
```diff
@@ -2,18 +2,45 @@ import { createTheme } from '@mui/material/styles';
   },
   palette: {
-    primary: { main: '#3A8DFF' },
+    primary: { main: '#f14140' },
```
👍🏽

rajivtitus commented on client/src/pages/SignUp/SignUp.tsx:
```diff
@@ -39,25 +39,18 @@ export default function Register(): JSX.Element {
             <SignUpForm handleSubmit={handleSubmit} />
           </Box>
-          <Box p={1} alignSelf="center" />
+          <AuthHeader linkTo="/login" asideText="Already a member?" linkText="Login" />
```
Can we rename this component to AuthFooter as we are using it to the bottom of the form?

rajivtitus reviewed (commented):
Great work with this! Just some minor comments to be address :)

MilesDev91 committed:
Cleanup styling

Moved font-weight styling to CSS instead of strong tags.

Replaced all-caps entries with text-transform in CSS.

Removed the global MUI Typography font-weight override in favor of using the useStyle files to override specific styling.

Renamed AuthHeader to AuthFooter and fixed all imports.

Added a .gitattributes file to help prevent git from converting files back to CRLF line endings.

MilesDev91 commented:
Fixed the concerns!

moffatethan reviewed (approved):
Good to go @MilesDev91 @rajivtitus 

🚀 💯
